### DISCUSSION TITLE
Generating Multiple drops from an item

### DISCUSSION THREAD

**Discussion by BSarmady** [misode/misode.github.io#380] [Q&A] [answered]
I like to drop multiple items from same block by random depending on conditions. after Reading very limited available sources I'm still not able to generate the desired outcome.

For example I like to modify Iron ore drop to be like following 

1- It always drop one cobble stone, or cobbled deepslate. it is silly that the cobble stone containing the iron ore suddenly disappears.
2- It always drops a random amount of Iron ore (what is default? 1?).
3- It drop only one of following items 
        a) by chance of 0.01 one gold ore
        b) by chance of 0.001 drop diamond
        c) by chance of 0.001 drop one emerald
4- If tool is silk touch then drop iron_ore itself

Also would be nice if I could only drop items in number 3 when diamond_pickaxe with fortune enchantment is used.

I tried to follow the wheat_seed and added multiple items to pools -> entries, but it sometimes drops more than one item from section 3 and sometimes It only drops just cobblestone or just Iron

Appreciate an example so I can follow and change my json file

**Comment by Nincodedo** [comment, marked as answer]
Adding multiple items to a pool will just give you more chances at that one roll. But adding multiple pools will allow multiple items to drop. So it would look something like this.

```
{
  "type": "minecraft:block",
  "pools": [
    {
      "rolls": 1,
      "bonus_rolls": 0,
      "entries": [
        {
          "type": "minecraft:alternatives",
          "children": [
            {
              "type": "minecraft:item",
              "name": "minecraft:iron_ore",
              "conditions": [
                {
                  "condition": "minecraft:match_tool",
                  "predicate": {
                    "enchantments": [
                      {
                        "enchantment": "minecraft:silk_touch",
                        "levels": {
                          "min": 1
                        }
                      }
                    ]
                  }
                }
              ]
            },
            {
              "type": "minecraft:item",
              "name": "minecraft:raw_iron",
              "functions": [
                {
                  "function": "minecraft:apply_bonus",
                  "enchantment": "minecraft:fortune",
                  "formula": "minecraft:ore_drops"
                },
                {
                  "function": "minecraft:explosion_decay"
                }
              ]
            }
          ]
        }
      ]
    },
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:cobblestone"
        }
      ]
    },
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "weight": 988,
          "name": "minecraft:stone"
        },
        {
          "type": "minecraft:item",
          "weight": 10,
          "name": "minecraft:raw_gold"
        },
        {
          "type": "minecraft:item",
          "name": "minecraft:emerald"
        },
        {
          "type": "minecraft:item",
          "name": "minecraft:diamond"
        }
      ]
    }
  ],
  "random_sequence": "minecraft:blocks/iron_ore"
}
```

**Reply by BSarmady** [reply]
Thank you very much, with some changes for my specific case, it worked.
